blues/reporters.py

```diff
@@ -0,0 +1,381 @@
+                data = data.encode('ascii')
+            if self.tables.__version__ >= '3.0.0':
+                self._handle.create_array(where='/', name='parameters', obj=[data])
+            else:
+                self._handle.createArray(where='/', name='parameters', object=[data])
+
+        if set_environment:
+            try:
+                envout = subprocess.check_output('conda env export --no-builds', shell=True, stderr=subprocess.STDOUT)
+                envjson = json.dumps(yaml.load(envout), sort_keys=True, indent=2)
+                if not isinstance(envjson, bytes):
+                    envjson = envjson.encode('ascii')
+                if self.tables.__version__ >= '3.0.0':
+                    self._handle.create_array(where='/', name='environment', obj=[envjson])
+                else:
+                    self._handle.createArray(where='/', name='environment', object=[envjson])
+            except Exception as e:
+                print(e)
+                pass
+
+class BLUESReporter(HDF5Reporter):
+    #This is a subclass of the HDF5 class from mdtraj that handles the reporting of
+    #the trajectory.
+    @property
+    def backend(self):
+        return BLUESHDF5TrajectoryFile
+
+    def __init__(self, file, reportInterval,
+                 title='NCMC Trajectory',
+                 coordinates=True, all_coordinates=False,
+                 time=False, cell=True, temperature=False,
+                 potentialEnergy=False, kineticEnergy=False,
+                 velocities=False, atomSubset=None,
+                 protocolWork=True, alchemicalLambda=True,
+                 parameters=None, environment=True):
+
+        super(BLUESReporter, self).__init__(file, reportInterval,
+            coordinates, time, cell, potentialEnergy, kineticEnergy,
+            temperature, velocities, atomSubset)
+
+        self._protocolWork = bool(protocolWork)
+        self._alchemicalLambda = bool(alchemicalLambda)
+        self._all_coordinates = bool(all_coordinates)
+        self._environment = bool(environment)
+        self._title = title
+        self._parameters = parameters
+
+    def report(self, simulation, state):
+        """Generate a report.
+        Parameters
+        ----------
+        simulation : simtk.openmm.app.Simulation
+            The Simulation to generate a report for
+        state : simtk.openmm.State
+            The current state of the simulation
+        """
+        if not self._is_intialized:
+            self._initialize(simulation)
+            self._is_intialized = True
+
+        self._checkForErrors(simulation, state)
+
+        args = ()
+        kwargs = {}
+        nsteps = simulation.integrator.getGlobalVariableByName('nsteps')
+        if self._coordinates:
+            coordinates = state.getPositions(asNumpy=True)[self._atomSlice]
+            coordinates = coordinates.value_in_unit(getattr(units, self._traj_file.distance_unit))
+            if not self._all_coordinates:
```

Comment:
Instead of the way the `all_coordinates` is currently set up, perhaps a more universal way to handle this would be to have it take a list/dict of ints, where each int corresponds to a frame of the trajectory. Then you can output trajectory frames if they are in that list/dict. This way you could also easily do things like output trajectory frames every X steps, for instance.